torch/cuda/memory.py

```diff
@@ -158,8 +158,8 @@ def empty_cache() -> None:
         of GPU memory in certain cases. See :ref:`cuda-memory-management` for
         more details about GPU memory management.
     """
-    if is_initialized():
-        torch._C._cuda_emptyCache()
+    _lazy_init()
+    torch._C._cuda_emptyCache()
```

Comment:
Why is this needed?  Won't the cache be empty if cuda is not initialized?

Reply:
`is_initialized` only checks python side. 

So we could have a python script that calls some C++ pybind, where only C++ does cuda operations. Once C++ is done, we want to clear the cache from python side, but this is_initialized() check will still be false, and we will silently ignore the call

Reply:
@xmfan Could you give a valid use case where a C++ extension is using device memory and PyTorch is still able to clear it without initializing a CUDA context?

Reply:
I don't think there is such a case, (before this PR) PyTorch doesn't initialize a CUDA context and silently ignores the empty_cache, so the C++ extension will continue to use device memory

This PR was mainly to address empty_cache silent failure, but had to be reverted due to being called from CPU-only code in depending libraries